cmake/extensions.cmake

```diff
@@ -585,6 +587,69 @@ function(board_finalize_runner_args runner)
     )
 endfunction()
 
+# 1.5. Misc.
+
+# zephyr_check_compiler_flag is a part of Zephyr's toolchain
+# infrastructure. It should be used when testing toolchain
+# capabilities and it should normally be used in place of the
+# functions:
+#
+# check_compiler_flag
+# check_c_compiler_flag
+# check_cxx_compiler_flag
+#
+# See check_compiler_flag() for API documentation as it has the same
+# API.
+#
+# It is implemented as a wrapper on top of check_compiler_flag, which
+# again wraps the CMake-builtin's check_c_compiler_flag and
+# check_cxx_compiler_flag.
+#
+# It takes time to check for compatibility of flags against toolchains
+# so we cache the capability test results in USER_CACHE_DIR (This
+# caching comes in addition to the caching that CMake does in the
+# build folder's CMakeCache.txt)
+function(zephyr_check_compiler_flag lang option check)
+  # Locate the cache
+  set_ifndef(
+    ZEPHYR_TOOLCHAIN_CAPABILITY_CACHE
+    ${USER_CACHE_DIR}/ToolchainCapabilityDatabase.cmake
+    )
+
+  # Read the cache
+  include(${ZEPHYR_TOOLCHAIN_CAPABILITY_CACHE} OPTIONAL)
```

Comment:
Is there a way to clear the cache?

Reply:
Since I understand this is stored in `USER_CACHE_DIR` and not the CMake cache itself, would this require a user to clear the %LOCALAPPDATA% folder if it upgrades the toolchain?

Reply:
This is kind of why I was thinking of emitting a message for where it's at -- gives users a sporting chance to delete it on their own if Zephyr doesn't provide means.

Reply:
"would this require a user to clear the %LOCALAPPDATA% folder if it upgrades the toolchain"

Assuming the caching mechanism works as intended, users will never have to clear the cache. Not even after toolchain upgrades. This is in part because the "key" in the cache includes the MD5sum of the CC binary.

Reply:
"This is kind of why I was thinking of emitting a message for where it's at -- gives users a sporting chance to delete it on their own if Zephyr doesn't provide means."

This has been introduced.